### PR TITLE
[imageinfo] Add new port

### PR DESCRIPTION
**Pull request by xiaozhuai**
**Describe the pull request**

[imageinfo](https://github.com/xiaozhuai/imageinfo) is a cross platform header only c++ library to get image size and format without loading/decoding. 
Supports avif, bmp, cur, dds, gif, hdr (pic), heic (heif), icns, ico, jp2, jpeg (jpg), jpx, ktx, png, psd, qoi, tga, tiff (tif), webp ...

### CODE REVIEW

File: ports/imageinfo/FindImageInfo.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# ^^^^^^^^^^^^^^^^
+#
+# This module defines the following variables:
+#
+# ``ImageInfo_FOUND``
+#   True if ImageInfo library found
+#
+# ``ImageInfo_INCLUDE_DIR``
+#   Location of ImageInfo headers
+#
```
Comment (JonLiu1993) [inline ports/imageinfo/FindImageInfo.cmake:19]: @xiaozhuai, Thanks for your PR, are these comments necessary? If not necessary, we recommend deleting these comments to keep the file as small as possible.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/imageinfo/FindImageInfo.cmake (hunk 1 above)
Comment (xiaozhuai) [reply ports/imageinfo/FindImageInfo.cmake:19]: Not necessary, I will update this pr.

**Comment by xiaozhuai** [comment]
@JonLiu1993 Please re-review.

File: ports/imageinfo/FindImageInfo.cmake [hunk 2]
```diff
@@ -0,0 +1,9 @@
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+include(${CMAKE_ROOT}/Modules/SelectLibraryConfigurations.cmake)
+
+if(NOT ImageInfo_INCLUDE_DIR)
+  find_path(ImageInfo_INCLUDE_DIR NAMES imageinfo.hpp PATHS ${ImageInfo_DIR} PATH_SUFFIXES include)
+endif()
+
+find_package_handle_standard_args(ImageInfo DEFAULT_MSG ImageInfo_INCLUDE_DIR)
+mark_as_advanced(ImageInfo_INCLUDE_DIR)
```
Comment (dg0yt) [inline ports/imageinfo/FindImageInfo.cmake:9]: Find Modules typically come with CMake or in user projects. They are needed when you need to implement a search procedure within the module.

But here, all information about the package is available at build time. In Modern CMake, packages should install Config files in this case. The config files has all information about a particular installation. (CMake adds a search procedure over multiple suitable configs.)

I would suggest the following:
- replace the find module with a Config file.
- derive ImageInfo_INCLUDE_DIR from the files location.

With the config file,
- you can drop the wrapper,
- you don't need find_package_handle_standard_args.

SelectLibraryConfigurations.cmake isn't needed anyways.

Normally, if the config file doesn't come with upstream's installation, it should be in the unofficial namespace for vcpkg.

File: ports/imageinfo/FindImageInfo.cmake [hunk 3]
```diff
@@ -0,0 +1,9 @@
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+include(${CMAKE_ROOT}/Modules/SelectLibraryConfigurations.cmake)
+
+if(NOT ImageInfo_INCLUDE_DIR)
+  find_path(ImageInfo_INCLUDE_DIR NAMES imageinfo.hpp PATHS ${ImageInfo_DIR} PATH_SUFFIXES include)
```
Comment (dg0yt) [inline ports/imageinfo/FindImageInfo.cmake:5]: `ImageInfo_DIR` doesn't have any useful meaning. And we know where the includes are, relative to `CMAKE_CURRENT_LIST_DIR`.

File: ports/imageinfo/FindImageInfo.cmake [hunk 4]
```diff
@@ -0,0 +1,9 @@
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+include(${CMAKE_ROOT}/Modules/SelectLibraryConfigurations.cmake)
+
+if(NOT ImageInfo_INCLUDE_DIR)
```
Comment (dg0yt) [inline ports/imageinfo/FindImageInfo.cmake:4]: This `if` guard is pointless. If `ImageInfo_INCLUDE_DIR` is set, `find_path` does nothing. This is the key pattern of caching in CMake.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by xiaozhuai** [comment]
@JonLiu1993 @dg0yt Please re-review

File: ports/imageinfo/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DIMAGEINFO_BUILD_TOOL=OFF
+        -DIMAGEINFO_BUILD_TESTS=OFF
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/imageinfo)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (dg0yt) [inline ports/imageinfo/portfile.cmake:19]: For header-only ports, I would suggest to add
~~~cmake
set(VCPKG_BUILD_TYPE release) # header-only port
~~~
before configure and install. There is no point in building the debug variant if it is completely removed.

File: ports/imageinfo/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,22 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO xiaozhuai/imageinfo
+    REF ea3eb17722a641f20729a9b0482898f7d8d4b07f # committed on 2023-01-28
+    SHA512 94ec6181878af05446c06ca70a55efcc7a712f5470a2cf177655a3c9fd95885dc4ce0213f1e4b5898eb5be47d0717713749b110a8d3416ad4409bc81c640ac98
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (dg0yt) [inline ports/imageinfo/portfile.cmake:10]: We would appreciate double quotes around `${SOURCE_PATH}`.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by xiaozhuai** [comment]
@dg0yt Done

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tested usage by `imageinfo:x64-windows` triplet successfully.

**Review by dg0yt — DISMISSED** [review]

**Review by BillyONeal — COMMENTED** [review]
Please just confirm that the usage text looks good and we'll land this for you. Thanks for your contribution to vcpkg!

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.  
  There is a repology entry for this which refers to http://www.wohlberg.net/public/software/photo/imageinfo/ but that seems like it has not changed since 2006, so I think it's OK for this to 'usurp' the name given that it's the first hit for a search for "imageinfo C++"
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.  
  Please confirm that this is correct. You can either just comment that it looks good and I'll add a usage file for you, or you can add one.
  ```
  imageinfo provides CMake targets:
  
      find_package(imageinfo CONFIG REQUIRED)
      target_link_libraries(main PRIVATE imageinfo::imageinfo)
  ```
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

**Comment by xiaozhuai** [comment]
> ```
> imageinfo provides CMake targets:
> 
>     find_package(imageinfo CONFIG REQUIRED)
>     target_link_libraries(main PRIVATE imageinfo::imageinfo)
> ```

That's looks good for me. Thank you.

**Review by BillyONeal — APPROVED** [review]
Thanks for adding the usage file for me @JonLiu1993

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Comment by xiaozhuai** [comment]
Thanks for all your guys! I am happy that this port can land.

**Pull request merged**